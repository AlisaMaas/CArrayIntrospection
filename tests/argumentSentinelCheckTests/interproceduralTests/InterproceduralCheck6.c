/**
 * This tests that we can't fool it by passing
 * a string literal in as the real null terminated
 * string argument of find and a possible null terminated
 * array as the non-string argument of find. We should find
 * that find's zeroth argument is a null-terminated string, the first
 * argument is not, and neither of foo's arguments is null-terminated.
 **/
void print(char *string) {}
int find(char string[], char nonString[]) {
	print(nonString);
	for (int i = 0;; i++) {
		if (nonString[i] != '\0') {
			print("About to break from loop");
			break;
		}
	}
	for (int i = 0;; i++) {
		if (string[i] == '\0') {
			print("About to break from loop");
			break;
		}
	}
	return 1;
}
int foo(char nonString[], char string[]) {
	print(nonString);
	for (int i = 0;; i++) {
		if (nonString[i] != '\0') {
			print("About to break from loop");
			break;
		}
	}
	return find("hello", string);
}
