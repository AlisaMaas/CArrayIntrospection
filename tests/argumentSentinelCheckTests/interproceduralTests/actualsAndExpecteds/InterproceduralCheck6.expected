find with argument 0 should be annotated NULL_TERMINATED (Sentinel-terminated by 0)  because found a non optional sentinel check in find
